opencog/nlp/sureal/SuRealSCM.cc

```diff
@@ -149,111 +149,14 @@ HandleSeqSeq SuRealSCM::do_sureal_match(Handle h, int thoroughness)
         sVars.insert(n);
     }
 
-    // separate the disconnected clauses (this will happen often with SuReal)
-    std::vector<HandleSeq> connectedClauses;
-    std::vector<std::set<Handle>> connectedVars;
-    get_connected_components(sVars, qClauses, connectedClauses, connectedVars);
-
-    logger().debug("[SuReal] Found %d disconnected components", connectedClauses.size());
-
-    std::map<Handle, std::vector<std::map<Handle, Handle> > > collector;
-
-    // call the pattern matcher on each set of disconnected commponents
-    for (size_t i=0; i<connectedClauses.size(); i++)
-    {
-        logger().debug("[SuReal] starting pattern matcher");
-        const HandleSeq& qClause(connectedClauses[i]);
-        const std::set<Handle>& qVars(connectedVars[i]);
-
-        // I replaced sVars by qVars in the below. sVars had extra
-        // variables that don't appear anywhere in the clauses -- linas.
-        SuRealPMCB pmcb(pAS, qVars, thoroughness);
-        PatternLinkPtr slp(createPatternLink(qVars, qClause));
-        slp->satisfy(pmcb);
-
-        // no pattern matcher result
-        if (pmcb.m_results.empty())
-            return HandleSeqSeq();
-
-        // first disconnected component & result? add it all
-        if (collector.empty())
-        {
-            collector = pmcb.m_results;
-            continue;
-        }
-
-        // if we are checking subsequent disconnected components, we want to
-        // make sure the new results can be merged with results from previously
-        // checked components; ie. only keep those with common
-        // InterpretationNode & no overlapping mappings
-        for (auto it = collector.begin(); it != collector.end(); )
-        {
-            // no common Interpretation, erase the old results as it can no
-            // longer be satisfied
-            if (pmcb.m_results.count(it->first) == 0)
-            {
-                logger().debug("[SuReal] Discarding a result for %s", it->first->toShortString().c_str());
-
-                it = collector.erase(it);
-                continue;
-            }
-
-            auto& existingMaps = it->second;                // a vector of previous mappings
-            auto& appendMaps = pmcb.m_results[it->first];   // a vector of unmerged mappings
-            std::vector<std::map<Handle, Handle> > newMaps;
-
-            // check all combinations of all the old mappings to the new
-            for (auto& em : existingMaps)
-            {
-                for (auto& am: appendMaps)
-                {
-                    // at this point, we know nothing in em & am would map the
-                    // same variable because they are from two disconnected
-                    // clauses.  Instead, we want to check to make sure no two
-                    // variables get mapping to the same node.
-                    auto checker = [&](const std::pair<Handle, Handle>& ekv)
```

Comment:
Hmm maybe not, but we also do this check this in grounding: https://github.com/leungmanhin/opencog/blob/su/opencog/nlp/sureal/SuRealPMCB.cc#L425